pkg/api/tracker.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"maps"
+	"sync"
+)
+
+// UserAgentTracker is a simple async hashmap that
+// tracks the reported User-Agent strings
+type UserAgentTracker struct {
```

Comment:
I don't think you need to make the UserAgentTracker.  You should be able to "GetorRegister" a counter where the name is the agent string, and use Inc(1) to add to it.  The use Get on the registry to retrieve it at the end of the test.  Also has the benefit that it will be printed when metrics are displayed.

Reply:
Did you have a particular implementation in mind? I was able to get it working like this:
```go
// Root handles / get requests
func (h *APIHandler) Root(w http.ResponseWriter, r *http.Request) {
	ua := useragent.Parse(r.Header.Get("User-Agent")).String

	rootcounter := metrics.NewCounter()
	rootcounter = h.userAgentsRoot.GetOrRegister(ua, rootcounter).(metrics.Counter)
	rootcounter.Inc(1)
```

Which feels kinda ugly to me?